src/commands/up.rs

```diff
@@ -127,10 +135,25 @@ impl UpCommand {
                     .as_deref()
                     .unwrap_or_else(|| DEFAULT_MANIFEST_FILE.as_ref());
                 let bindle_connection = self.bindle_connection();
-                spin_loader::from_file(manifest_file, &working_dir, &bindle_connection).await?
+                spin_loader::from_file(
+                    manifest_file,
+                    if self.direct_mounts {
+                        None
+                    } else {
+                        Some(&working_dir)
+                    },
+                    &bindle_connection,
+                )
+                .await?
             }
             (None, Some(bindle)) => match &self.server {
-                Some(server) => spin_loader::from_bindle(bindle, server, &working_dir).await?,
+                Some(server) => {
+                    if self.direct_mounts {
```

Comment:
If we get clap to block the combination as @etehtsea suggested, then could we remove this check?  Belt and braces, yes, ideally, but there's already a lot of nesting of conditions (combined with other indents that obscure the structure, THANKS CARGO FMT), and this additional level starts to make the code very in-and-out-y if that makes sense...